how-to-guides/sagemaker/pytorch/Using-Neptune-with-PyTorch.ipynb

```diff
@@ -0,0 +1,234 @@
+    "print(\"Test data:\")\n",
+    "!aws s3 ls --recursive $test_s3uri"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6c1caf03",
+   "metadata": {},
+   "source": [
+    "## Train \n",
+    "\n",
+    "We are going to use the [SageMakers PyTorch Estimator](https://sagemaker.readthedocs.io/en/stable/frameworks/pytorch/sagemaker.pytorch.html).\n",
+    "The estimator uses the code from the `code/` directory. The code is adapted from the [AWS SageMaker Examples repository](https://github.com/aws/amazon-sagemaker-examples/tree/main/advanced_functionality/multi_model_pytorch).\n",
+    "What was changed in the code is that Neptune logging was added to the `code/train.py` script.\n",
+    "\n",
+    "```diff\n",
+    "[...]\n",
+    "\n",
+    "def train(args):\n",
+    "+   run = neptune.init_run(tags=[\"sagemaker\"])\n",
+    "    \n",
+    "    [...]\n",
+    "    \n",
+    "+   run[\"training/args\"] = args\n",
+    "+   run[\"training/model/loss_fn\"] = type(loss_fn).__name__\n",
+    "+   run[\"training/model/model\"] = type(net).__name__\n",
+    "+   run[\"training/model/optimizer\"] = type(optimizer).__name__\n",
+    "\n",
+    "    logger.info(\"Start training ...\")\n",
+    "    for epoch in range(1, args.epochs + 1):\n",
+    "        net.train()\n",
+    "        for batch_idx, (imgs, labels) in enumerate(train_loader, 1):\n",
+    "            imgs, labels = imgs.to(device), labels.to(device)\n",
+    "            output = net(imgs)\n",
+    "            loss = loss_fn(output, labels)\n",
+    "\n",
+    "            optimizer.zero_grad()\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "            [...]\n",
+    "            \n",
+    "+           run[\"training/train/batch/loss\"].log(loss.item())\n",
+    "\n",
+    "        # test the model\n",
+    "+       train_loss, train_acc = test(net, train_loader, device)\n",
+    "+       run[\"training/train/epoch/loss\"].log(train_loss)\n",
+    "+       run[\"training/train/epoch/accuracy\"].log(train_acc)\n",
+    "        \n",
+    "+       test_loss, test_acc = test(net, test_loader, device)\n",
+    "+       run[\"training/test/epoch/loss\"].log(test_loss)\n",
+    "+       run[\"training/test/epoch/accuracy\"].log(test_acc)\n",
+    "\n",
+    "[...]\n",
+    "        \n",
+    "```\n",
+    "\n",
+    "Another difference is that we need `neptune-client` as an additional dependency, so we add it to `code/requirements.txt`. The SageMaker Estimator by default installs all the dependencies defined there.\n",
+    "\n",
+    "Because AWS SageMaker does not use python>=3.6 by default, we need to use a custom docker image to train the model as python3.6 has already reached end-of-life and is not supported by Neptune. For this purpose, we use [one of the images provided by AWS](https://github.com/aws/deep-learning-containers/blob/master/available_images.md) as `image_uri`. \n",
```

Comment:
```suggestion
    "Because SageMaker does not use python>=3.6 by default, we need to use a custom Docker image to train the model as Python 3.6 has already reached end-of-life and is not supported by Neptune. For this purpose, we use [one of the images provided by AWS](https://github.com/aws/deep-learning-containers/blob/master/available_images.md) as `image_uri`. \n",
```